### PR TITLE
feat: Form GPA and Graduation Year Validation

### PR DESCRIPTION
**Pull request by wderocco8**
## What does this PR do?

Refactored applicant Form.tsx to ensure valid inputs that will integrate more easily with data-analytics. Specifically, the `gpa` field is now required to be a floating point number between 0 and 4. Also added a `gradMonth` field that is separate from `gradYear` (which now must be an integer).

## Type of change

- [ ] Fix: Bug fix (non-breaking change which fixes an issue)
- [x] Refactor: Any code refactoring
- [ ] Chore: technical debt, workflow improvements
- [ ] Feature: New feature (non-breaking change which adds functionality)
- [ ] Documentation: This change requires a documentation update

## Tests Performed

Performed the following tests on the frontend
1. Submit form with `n/a` gpa
2. Submit form with valid numeric gpa
3. Attempt to submit form without valid numeric gpa (fails -> good)
4. Submit form with valid month/year

## Screenshots

image

### CODE REVIEW

File: app/admin/listing/[listingId]/insights/page.tsx [hunk 1]
```diff
@@ -22,7 +31,7 @@ export default function Insights({ params }: { params: { listingId: string } })
     linkedin: [],
     website: [],
   });
-  
+  console.log(applicantData)
```
Comment (jinyoungbang) [inline app/admin/listing/[listingId]/insights/page.tsx:34]: Just a good idea to remove general data-related console logs!

**Review by jinyoungbang — CHANGES_REQUESTED** [review]
Since this is a public-related feature, my comments might be a little bit more strict 🥲

- GPA section: the styling and UX is a bit awkward – let's try brainstorm some other ideas
- I love the idea for graduation year and month being separated! Two concerns: 1. When we send our formData (through `handleSubmit`), the format of our data will be changing so I'm wondering if there's a way to make that consistent like before. 2. Could we change the graduation month to a dropdown with two options (January and May)? I think that UX is more intuitive for the user and we wouldn't need to bother with validation.
- The reason why we didn't put what fields are missing on alert is because it notifies the user the id of what field is missing, which I would think lead to poor UX. I think providing info to the user is very valuable, but UX would come first for this case.
- The `renderGpaSection()` and rendering the graduation date is missing the required attribute on the tag (not too sure how it'd work with split values)
- Finally, the dropdown on GPA I think is not intuitive as clicking on the arrow allows us  increment/decrement integers as a whole (not floats). If you can find a fix to increment gpa by +/- 0.01, give it a shot!

Overall, amazing initiative and some really great ideas! Feel free to send me another review whenever :)

giphy

File: app/admin/listing/[listingId]/insights/page.tsx (hunk 1 above)
Comment (wderocco8) [reply app/admin/listing/[listingId]/insights/page.tsx:34]: valid, forgot i added lol

**Comment by wderocco8** [comment]
> Since this is a public-related feature, my comments might be a little bit more strict 🥲
> 
> * GPA section: the styling and UX is a bit awkward – let's try brainstorm some other ideas
> * I love the idea for graduation year and month being separated! Two concerns: 1. When we send our formData (through `handleSubmit`), the format of our data will be changing so I'm wondering if there's a way to make that consistent like before. 2. Could we change the graduation month to a dropdown with two options (January and May)? I think that UX is more intuitive for the user and we wouldn't need to bother with validation.
> * The reason why we didn't put what fields are missing on alert is because it notifies the user the id of what field is missing, which I would think lead to poor UX. I think providing info to the user is very valuable, but UX would come first for this case.
> * The `renderGpaSection()` and rendering the graduation date is missing the required attribute on the tag (not too sure how it'd work with split values)
> * Finally, the dropdown on GPA I think is not intuitive as clicking on the arrow allows us  increment/decrement integers as a whole (not floats). If you can find a fix to increment gpa by +/- 0.01, give it a shot!
> 
> Overall, amazing initiative and some really great ideas! Feel free to send me another review whenever :)
> 
> giphy [ giphy ](https://private-user-images.githubusercontent.com/44107246/298355901-2de1a907-298a-48fc-b190-27e86a0ebce9.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU4MTI2NzAsIm5iZiI6MTcwNTgxMjM3MCwicGF0aCI6Ii80NDEwNzI0Ni8yOTgzNTU5MDEtMmRlMWE5MDctMjk4YS00OGZjLWIxOTAtMjdlODZhMGViY2U5LmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTIxVDA0NDYxMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFlZGYzNTBlZTViOTdiYTBkOTFhNjJjYjQ5MjRkZTY1Njc1MzcxOWZhOTUyMjRlNzM2YzlhYjk1YWYzYTBhYjUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.ziirSqdkLcd1VwhZgqs4iFPlrF462rk9-D18NSKyV3I) [ ](https://private-user-images.githubusercontent.com/44107246/298355901-2de1a907-298a-48fc-b190-27e86a0ebce9.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU4MTI2NzAsIm5iZiI6MTcwNTgxMjM3MCwicGF0aCI6Ii80NDEwNzI0Ni8yOTgzNTU5MDEtMmRlMWE5MDctMjk4YS00OGZjLWIxOTAtMjdlODZhMGViY2U5LmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTIxVDA0NDYxMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFlZGYzNTBlZTViOTdiYTBkOTFhNjJjYjQ5MjRkZTY1Njc1MzcxOWZhOTUyMjRlNzM2YzlhYjk1YWYzYTBhYjUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.ziirSqdkLcd1VwhZgqs4iFPlrF462rk9-D18NSKyV3I)

**Review by jinyoungbang — APPROVED** [review]

**Pull request merged**